docs/website/contents/for-developers/HardForks.md

```diff
@@ -0,0 +1,98 @@
+
+It's ultimately the judgment of the node developers whether some changes should be achieved with an era transition rather than an intra-era hard fork.
+
+The Cardano developers have so far used very expressive static typing.
+It's plausible that an implementation that instead used dynamic types could more frequently use intra-era hard forks rather than era transitions.
+However, we strongly believe that overall development and assurance would degrade without the benefits of such precise types.
+
+The first era transition (from Byron to Shelley) is a notable example.
+It was the first era transition we experienced, and no subsequent transition changes nearly so much.
+The two code bases are drastically different, so it's hard to argue this should have been an intra-era hard fork.
+In particular, this is the only era transition that changed the duration of slots and the number of slots in an epoch.
+In the current codebase, any hard fork that changes those per-era constants must be an era transition, since the Consensus infrastructure was explicitly designed with that restriction as a simplifying assumption.
+A different design might relax that constraint, but it is has not been restrictive so far: a second change to these constants has been discussed only rarely and never actually proposed, and deferring them until whatever justifies the next era transition wouldn't obviously be prohibitively disruptive.
+
+### How wide-reaching are eras?
+
+Eras are ultimately an implementation detail of the node, a consequence of architectural trade-offs.
+The node's behaviors currently reflects the eras in only two ways.
+
+Specifically, the node must be able to forecast the data necessary to validate a useful prefix of some header chain from the intersection of that header chain and the node's current selection, including whether each header arrived too early.
+
+- The familiar and intuitive notion of era is useful for organization in general, and so the eras' names might appear within parameter names in configuration files, specification documents, announcements, etc.
+
+  However, the era transitions themselves should generally be considered as "just a big hard fork".
+  The foundational data is instead the protocol version, especially because not all hard forks are era transitions.
+  (See [this GitHub comment](https://github.com/IntersectMBO/ouroboros-consensus/issues/416#issuecomment-2669347315) for the details of how this lesson was learned.)
+
+- Some codecs reflect Cardano's specific era structure.
+  That's  not strictly necessary, but a tagged union approach has made it trivial to combine the era-specific codecs into a full Cardano codec.
+  Specifically, top-level codecs fundamentally branch on a simple number located near the beginning of the bytestring and dispatch accordingly to the corresponding era's codecs.
+
+- (TODO Confirm whether these tags dont affect any hashes. If they do, to what extent does it actually matter?)
```

Comment:
I am not aware of any tags. As such no hashes are affected.

Reply:
Ha! Proof by "I've never heard of that" :D Excellent.